### ISSUE TITLE
RuntimeError on ONNX export of `fake_quantize_per_tensor_affine`

### ISSUE THREAD

**Issue by bfineran** [pytorch/pytorch#77455]
### 🐛 Describe the bug

On `torch.onnx` export of a quantized model, the following error occurs on recent pytorch versions (>1.9).  The model is quantized using `fake_quantize_per_tensor_affine` with the supported `prepare_qat_` flows.  This issue does not exist in torch versions `<= 1.9`.  It is worth noting that I'm seeing slightly different issues on nightly and the 1.11 release, but both seem related to reading the scale from `fake_quantize_per_tensor_affine`.

It looks like another user ran into the same issue here https://github.com/pytorch/pytorch/issues/39502#issuecomment-972558864, but am opening a main issue since it looks like that thread has gone inactive.

Stack trace (torch 1.11.0):
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "...", line 470, in export_onnx
    torch.onnx.export(
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/__init__.py", line 305, in export
    return utils.export(model, args, f, export_params, verbose, training,
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/utils.py", line 118, in export
    _export(model, args, f, export_params, verbose, training, input_names, output_names,
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/utils.py", line 719, in _export
    _model_to_graph(model, args, verbose, input_names,
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/utils.py", line 503, in _model_to_graph
    graph = _optimize_graph(graph, operator_export_type,
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/utils.py", line 232, in _optimize_graph
    graph = torch._C._jit_pass_onnx(graph, operator_export_type)
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/__init__.py", line 354, in _run_symbolic_function
    return utils._run_symbolic_function(*args, **kwargs)
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/utils.py", line 1061, in _run_symbolic_function
    return symbolic_fn(g, *inputs, **attrs)
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/symbolic_helper.py", line 166, in wrapper
    args = [_parse_arg(arg, arg_desc, arg_name, fn_name)  # type: ignore[assignment]
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/symbolic_helper.py", line 166, in <listcomp>
    args = [_parse_arg(arg, arg_desc, arg_name, fn_name)  # type: ignore[assignment]
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/symbolic_helper.py", line 93, in _parse_arg
    raise RuntimeError("Expected node type 'onnx::Constant' "
RuntimeError: Expected node type 'onnx::Constant' for argument 'scale' of node 'fake_quantize_per_tensor_affine', got 'prim::Param'.
```

stack trace (nightly):
``` 
File "...", line 470, in export_onnx
    torch.onnx.export(
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/__init__.py", line 350, in export
    return utils.export(
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/utils.py", line 148, in export
    _export(
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/utils.py", line 1082, in _export
    graph, params_dict, torch_out = _model_to_graph(
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/utils.py", line 724, in _model_to_graph
    graph = _optimize_graph(
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/utils.py", line 300, in _optimize_graph
    graph = torch._C._jit_pass_onnx(graph, operator_export_type)
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/__init__.py", line 416, in _run_symbolic_function
    return utils._run_symbolic_function(*args, **kwargs)
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/utils.py", line 1561, in _run_symbolic_function
    return symbolic_fn(g, *inputs, **attrs)
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/symbolic_helper.py", line 227, in wrapper
    return fn(g, *args, **kwargs)
  File ".../test-env/lib/python3.8/site-packages/torch/onnx/symbolic_opset10.py", line 427, in fake_quantize_per_tensor_affine
    scale = scale.float().data  # Avoid exporter generating double type
AttributeError: 'torch._C.Value' object has no attribute 'float'
```

### Versions

stack trace from torch 1.11.0,  second trace is from torch nightly (`torch-1.12.0.dev20220513+cpu torchvision-0.13.0.dev20220513+cpu`)

results of `collect_env.py`:
```
Collecting environment information...
PyTorch version: 1.11.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.18.0-rc3
Libc version: glibc-2.27

Python version: 3.8.0 (default, Feb  8 2021, 21:05:07)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.0-105-generic-x86_64-with-glibc2.27
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: NVIDIA TITAN RTX
Nvidia driver version: 470.103.01
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.22.1
[pip3] torch==1.11.0
[pip3] torchvision==0.12.0
[conda] Could not collect
```

**Comment by BowenBao** [comment]
Tracked internally by Microsoft.

@bfineran We are adding more support to quantized model export. It looks like the problem with nightly is that the scale value is dynamic where as the exporter considers only static scale at the moment.

Could you share a small minimum repro example for us to take a look? Thank you.

**Comment by bfineran** [comment]
hi @BowenBao thank you for the response, and sorry for the delay.  I believe the model I was using was using uses static quantization, going through default qat flows.  Here's a minimally reproducible example below - note that disabling the observers is the change that leads to this specific error, but another error will also be given if observers are still active.

```python
import torch

model = torch.nn.Linear(10, 10)
model.qconfig = torch.ao.quantization.qconfig.default_qat_qconfig
torch.ao.quantization.prepare_qat(model, inplace=True)

_ = model(torch.randn(10, 10))
model.apply(torch.ao.quantization.disable_observer)

torch.onnx.export(model, torch.randn(10, 10), "qat-error-example.onnx", opset_version=11)
```

**Comment by bfineran** [comment]
@BowenBao looks like upgrading to latest nightly (>1.13) may have resolved this issue, however if you have any insight on what changes could help resolve the issue for any more current torch versions (1.10-1.12), I would greatly appreciate that

**Comment by markurtz** [comment]
Hi @BowenBao is there any update on this? It's limiting for a lot of repos currently for using quantization.

**Comment by BowenBao** [comment]
Hi @bfineran, @markurtz , apologize for the late response. `fake_quantize_per_tensor_affine` should have been supported via https://github.com/pytorch/pytorch/pull/76055. As @bfineran mentioned, it should work for latest nightly.

**Comment by markurtz** [comment]
Hi @BowenBao , thanks for the reply. Given that quantized model export is broken in 1.9 through 1.12, is there anyway to get the fix cherry picked into big release for any of those? Nightly isn't a stable option for most people for production needs.

**Comment by BowenBao** [comment]
Hi @markurtz, I think above script works under 1.12, `opset_version` needs to be changed to 13 since it depends on a later ONNX operator version. That being said, ONNX quantization support is fairly new and we are actively working on it. Please use nightly to validate export on your quantization model, as it has the latest support. We will try our best to pack the newer fixes to the next minor release.

**Comment by BowenBao** [comment]
Above mentioned fixes were packed in release 1.12.1. More support are coming in the next 1.13 release and further. Closing
